### ISSUE TITLE
Add "vice-county (supplied)" as a configurable column option to the My Records grid

### ISSUE THREAD

**Issue by sacrevert** [BiologicalRecordsCentre/iRecord#1741]
Required for the field to be useful within My Records

**Comment by johnvanbreda** [comment]
There's a few hours work required here, as the supplied VC is not currently in the Elasticsearch index (only the stored integer location ID). Please confirm if this is worth doing.

**Comment by sacrevert** [comment]
I think it's worth doing for BBS data management: if we are asking VCRs to move to iRecord and importing historic data from our BRC Oracle database, then folk need to be able to sort their data by the supplied VC. Ofc, I recognise that there are resourcing issues that @kitenetter and @DavidRoy probably need to comment on

**Comment by DavidRoy** [comment]
I agree. Please go ahead @johnvanbreda with existing 'BRC' budgets for iRecord support and development. This is of use for multiple schemes

**Comment by johnvanbreda** [comment]
Done in the code for the next release, e.g. there will be a column called "Supplied region location code" and one called "Supplied region location name"

**Comment by sacrevert** [comment]
@johnvanbreda sorry to be a pain here, but is  "Supplied region location code" == region database ID code and "Supplied region location name" == the vice-county number where the region in question is a vice-county?

**Comment by johnvanbreda** [comment]
Not being a pain at all - the code will be the VC number, the name will be the full name. There is also a Supplied region location ID for the database ID if you need it. Note that the default wording needs to be generic as Indicia can be configured for this to be a NUTS region or whatever, depending on the project, so not always a Vice County.

**Comment by sacrevert** [comment]
Thanks, I guessed the reason for the generic-sounding label; however, given that the BBS and other societies interacting with this through My Records will be using it for vice-county information, can it be relabelled on presentation here as "vice-county number (supplied)" and "vice-county name (supplied)". The main reason for the BBS paying for this was so that they could easily organise their existing data on a vice-county basis, and the main route for this will be My Records and related download options (because that is presumably the way that a VCR would access their vice-county holdings). So, using an obscure name for the field at this point of access is not terribly helpful...

**Comment by DavidHepper** [comment]
BDS has a not-very-important custom field 'VC' to contain the verbatim VC number entered by the original recorder. In due course should we change to using these new field(s) or is it safer to leave well alone?

**Comment by johnvanbreda** [comment]
@sacrevert I've added a language override so that iRecord will use terms like "Vice-county location name (supplied)" when this gets deployed.

@DavidHepper it's fine to leave well alone. The advantage of a switch would be that if a user specifies a VC using the new field type then the record gets allocated to that VC for verification even when it overlaps. There is code on the page asking the user to select one of the overlapping counties but this is only triggered if the record overlaps. Whereas your current VC field has no additional functionality.

**Comment by johnvanbreda** [comment]
Now deployed.

**Comment by sacrevert** [comment]
This information does not seem to be being displayed on the My Records grid:

![Image](https://github.com/user-attachments/assets/c3bdc295-583c-46ae-af56-d088ed37510f)

These records were uploaded under training mode, and the account is still in training mode for this view. Despite vice-county numbers being matched to `Sample attributes for selected survey :: Vice-county number` during the import, nothing is being displayed in the My Records grid.

**Comment by johnvanbreda** [comment]
The correct field to import against in this scenario was the Vice County (lookup using location code). The Vice-county number field is just a plain integer without additional functionality as it maps to Vice-county (supplied), attribute ID 256.

**Comment by sacrevert** [comment]
thanks @johnvanbreda that's entirely my fault then as I have been telling people the wrong information :)

I will close this and create another issue relating to fixing all the errors that I have therefore presumably caused here.
